xds/src/main/java/io/grpc/xds/EdsLoadBalancer.java

```diff
@@ -321,7 +287,26 @@ public int hashCode() {
         EdsLoadBalancer.this.endpointWatcher = endpointWatcher;
       }
 
-      // TODO(zddapeng): In handleResolvedAddresses() handle child policy change if any.
+      @Override
+      public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+        XdsConfig config = (XdsConfig) resolvedAddresses.getLoadBalancingPolicyConfig();
+        if (config.lrsServerName != null) {
+          if (!config.lrsServerName.equals("")) {
+            throw new AssertionError(
+                "Can only report load to the same management server");
+          }
+          if (!isReportingLoad) {
+            xdsClient.reportClientStats(clusterName, null, loadStatsStore);
```

Comment:
This only works for the case cluster_service name is never changed, otherwise it will throw because of the graceful switch. It that intended behavior for now?